rllib/algorithms/appo/tf/appo_tf_rl_module.py

```diff
@@ -49,8 +46,10 @@ def output_specs_train(self) -> List[str]:
     @override(PPOTfRLModule)
     def _forward_train(self, batch: NestedDict):
         outs = super()._forward_train(batch)
-        old_pi_inputs_encoded = self.old_encoder(batch)[ENCODER_OUT][ACTOR]
-        old_action_dist_logits = self.old_pi(old_pi_inputs_encoded)
+        old_pi_inputs_encoded = tf.stop_gradient(
+            self.old_encoder(batch)[ENCODER_OUT][ACTOR]
+        )
+        old_action_dist_logits = tf.stop_gradient(self.old_pi(old_pi_inputs_encoded))
```

Comment:
if old_pi_inputs_encoded aren't used anywhere else besides here, then a tf.stop_gradient on this line will suffice instead of having to do it in both places.

Reply:
You are right. Fixed.